From 8c2a436addfa38d3189f5ab41f6dad961bda4e59 Mon Sep 17 00:00:00 2001
From: Les Lee <lesl@google.com>
Date: Mon, 22 Aug 2022 15:36:52 +0800
Subject: [PATCH] [DO NOT MERGE] wifi: Reset to default SAP configuration when
 doing factory reset

Bug: 241927115
Test: atest -c FrameworksWifiTests
Change-Id: Ifdab8dd4bfec9012ed5c052e1b468aacc7cf9749
Merged-In: Ia212666450fd9fe3a95393defa2fb56383bab83a
(cherry picked from commit 08358dfa26ea71114a2c1e4817aae8a21350d38e)
Merged-In: Ifdab8dd4bfec9012ed5c052e1b468aacc7cf9749
---
 service/java/com/android/server/wifi/WifiServiceImpl.java      | 3 +++
 .../src/com/android/server/wifi/WifiServiceImplTest.java       | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/service/java/com/android/server/wifi/WifiServiceImpl.java b/service/java/com/android/server/wifi/WifiServiceImpl.java
index 4b570365c..bb72876bb 100644
--- a/service/java/com/android/server/wifi/WifiServiceImpl.java
+++ b/service/java/com/android/server/wifi/WifiServiceImpl.java
@@ -3462,6 +3462,9 @@ public class WifiServiceImpl extends BaseWifiService {
             removePasspointConfigurationInternal(null, config.getUniqueId());
         }
         mWifiThreadRunner.post(() -> {
+            EventLog.writeEvent(0x534e4554, "241927115", -1,
+                    "Reset SoftApConfiguration to default configuration");
+            mWifiApConfigStore.setApConfiguration(null);
             mPasspointManager.clearAnqpRequestsAndFlushCache();
             mWifiConfigManager.clearUserTemporarilyDisabledList();
             mWifiConfigManager.removeAllEphemeralOrPasspointConfiguredNetworks();
diff --git a/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java b/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
index 87e09ac07..cae3c7b2b 100644
--- a/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
+++ b/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
@@ -4216,7 +4216,7 @@ public class WifiServiceImplTest extends WifiBaseTest {
 
         // Let the final post inside the |factoryReset| method run to completion.
         mLooper.dispatchAll();
-
+        verify(mWifiApConfigStore).setApConfiguration(null);
         verify(mWifiConfigManager).removeNetwork(
                 openNetwork.networkId, Binder.getCallingUid(), TEST_PACKAGE_NAME);
         verify(mWifiConfigManager).removeNetwork(
-- 
2.39.0.rc1.256.g54fd8350bd-goog

